tests/textcase/plugin/commands/text_case_start_replacing_with_first_part_command_spec.lua

```diff
@@ -0,0 +1,21 @@
+local textcase = require("textcase")
+local test_helpers = require("tests.test_helpers")
+
+describe("plugin commands TextCaseStartReplacingWithFirstPartCommand", function()
+  before_each(function()
+    textcase.setup()
+    vim.api.nvim_set_keymap("n", "gar", "<cmd>TextCaseStartReplacingWithFirstPartCommand<CR>", {})
+
+    local buf = vim.api.nvim_create_buf(false, true)
+    vim.api.nvim_command("buffer " .. buf)
+    vim.api.nvim_buf_set_lines(0, 0, -1, true, { "LoremIpsum DolorSit" })
+  end)
+
+  describe("TextCaseStartReplacingWithFirstPartCommand", function()
+    it("should select the first part of the current word for the Subs command", function()
```

Comment:
What is the use?

Reply:
If I have `CarList`, I want to open the command line with `Subs/Car/`

Reply:
How about, a parametrizable command that receives start and amount of words.

Let's say the user wants to change 3 words starting on the second, they should be able to create a keybinding like
{"<leader>i", "<cmd>TextCaseStartReplacingSubpart 2 3<CR>"}

TextCaseStartReplacingSubpart is a bad name xD. only used as an example

Reply:
For documentation purpose and variable names, how are these small particles of word called?

Reply:
Oh no this turned out to be way more than I expected when I thought about this :D

It would definitely be nice if it was customizable.

So for example (x can be anything of course)
- Cursor on `Car` in `CarList`, `gax` => `Subs/Car`
- Cursor on `Car` in `EnhancedCarList`, `gax` => `Subs/Car`
- Cursor on `Enhanced` in `EnhancedCarList`, `ga2x` => `Subs/EnhancedCar`

Not sure if this is possible.

Reply:
I mean, first of all, I think this is a great great idea!!, I can see myself using it for many different things and It does not have to be solved in the short term

Since every new command we document will be a commitment to avoid breaking changes, we can start by hard coding your current implementation

TextCaseStartReplacingParts, receiving 2 parameters. And hard coded to work with only the first part. wdyt?

Reply:
On a second thought, it is not documented and not public API yet xD
Maybe merging it with a commit note specifying it is work in progress for x feature

Reply:
Yeah, could be marked as alpha in the code even.

Reply:
But now that I think about it does it even need to be a command? It could also just be a lua function, right?

Reply:
I will have another look at it tomorrow and think more about it.

Reply:
Ah good idea, so that it is not exposed. then in your local you can 

`:lua require('textcase').method_name()`